sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala

```diff
@@ -65,6 +65,38 @@ object StringUtils extends Logging {
     "(?s)" + out.result() // (?s) enables dotall mode, causing "." to match new lines
   }
 
+  def escapeLikeRegex(pattern: String, escapeStr: String): String = {
```

Comment:
Why we need the two version of `escapeLikeRegex`? Can we merge them?

Reply:
Yes, the old `escapeLikeRegex` will not be used. I will remove it.